crates/bevy_time/src/commands.rs

```diff
@@ -0,0 +1,186 @@
+use std::{
+    ops::{Deref, DerefMut},
+    time::Duration,
+};
+
+use bevy_ecs::{
+    system::{Commands, Deferred, Res, ResMut, Resource, SystemBuffer, SystemMeta, SystemParam},
+    world::{CommandQueue, World},
+};
+
+use crate::{Domain, Time, Timer, TimerMode};
+
+/// A [`SystemParam`] that allows you to queue commands to be
+/// executed after a certain duration has elapsed,
+/// rather than immediately at the next sync point.
+///
+/// This type also derefs to [`Commands`], so you can still use it
+/// to queue commands to run at the next sync point.
+///
+/// **Note**: By default, commands that have elapsed will be queued
+/// in the [`First`] and [`FixedFirst`] schedules for time domains
+/// `()` and [`Fixed`], respectively.
+///
+/// # Usage
+///
+/// Add `mut commands: TimedCommands` as a function argument to your system,
+/// and call [`TimedCommands::after`] to get a [`Commands`] that will queue its commands
+/// to run after the specified duration has elapsed.
+///
+/// ```
+/// # use bevy_time::prelude::*;
+/// # use std::time::Duration;
+/// #
+/// fn my_system(mut commands: TimedCommands) {
+///     commands.after(Duration::from_secs(5))
+///         .spawn_empty();
+///
+///     commands.spawn_empty();
+/// }
+/// # bevy_ecs::system::assert_is_system(my_system);
+/// ```
+///
+/// [`First`]: bevy_app::First
+/// [`FixedFirst`]: bevy_app::FixedFirst
+/// [`Fixed`]: crate::Fixed
+#[derive(SystemParam)]
+pub struct TimedCommands<'w, 's, T: Domain = ()> {
+    commands: Commands<'w, 's>,
+    queues: Deferred<'s, TimedCommandQueues<T>>,
+}
```

Comment:
I think if `Time` was updated to store the `TypeId` of the original domain (e.g., during calls to `.as_generic()`), we could take `time: Res<Time<T>>,` as an additional parameter here, and use it to contextualize `Time<()>`. It still wouldn't help with custom time domains though.